docs/examples.md

```diff
@@ -337,3 +337,43 @@ s.on('end', function () {
 
     $ node line_count.js count -f line_count.js
     25
+
+Using inquirer for prompting
+---------------------------
+
+```js
+const inquirer = require('inquirer');
+const yargs = require('yargs');
+
+const askName = async () => {
+  const answers = await inquirer.prompt([
+    {
+      message: 'What is your name?',
+      name: 'name',
+      type: 'string'
+    }
+  ]);
+
+  console.log(`Hello, ${answers.name}!`);
+};
+
+const argv = yargs(process.argv.splice(2))
+  .command('ask', 'use inquirer to prompt for your name')
```

Comment:
rather than using a switch statement, you could do this:

```js
.command('ask', 'use inquirer to prompt for your name', () => {}, async () => {
   await askName();
}
```

Reply:
ok, it's working without the `async/await` though

```js
.command('ask', 'use inquirer to prompt for your name', () => {}, askName)
```

Reply:
👍

Reply:
the advantage of async await is that you could `await` the top level `yargs.argv` and it shouln't return until you've filled in the prompted messages ... neat right?

Reply:
actually I've first tested with your suggestion, and it was not working as expected, not sure why;